src/fast_array_utils/stats/_sum.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: MPL-2.0
+from __future__ import annotations
+
+from functools import singledispatch
+from typing import TYPE_CHECKING, Any, overload
+
+import numpy as np
+
+from ..types import DaskArray
+
+
+if TYPE_CHECKING:
+    from typing import Literal, TypeVar
+
+    from numpy.typing import NDArray
+
+    from ..types import CSBase, CSMatrix
+
+    DT_co = TypeVar("DT_co", covariant=True, bound=np.generic)
+
+    _Ax = tuple[Literal[0, 1], ...] | Literal[0, 1]
+
+
+@overload
+def sum(
+    x: NDArray[DT_co] | CSBase[DT_co], *, axis: _Ax | None = None, dtype: None = None
+) -> NDArray[DT_co]: ...
+
+
+@overload
+def sum(
+    x: NDArray[Any] | CSBase[Any], *, axis: _Ax | None = None, dtype: np.dtype[DT_co]
+) -> NDArray[DT_co]: ...
+
+
+@singledispatch
+def sum(
+    x: NDArray[DT_co] | CSBase[DT_co],
+    *,
+    axis: _Ax | None = None,
+    dtype: np.dtype[DT_co] | None = None,
+) -> NDArray[DT_co]:
+    return np.sum(np.asarray(x), axis=axis, dtype=dtype)
+
+
+@sum.register(DaskArray)
+def _(x: DaskArray, *, axis: _Ax | None = None, dtype: np.dtype[DT_co] | None = None) -> DaskArray:
+    import dask.array as da
+
+    # TODO(@ilan-gold): why is this so complicated?
+    # https://github.com/scverse/scanpy/pull/2856/commits/feac6bc7bea69e4cc343a35855307145854a9bc8
+    if dtype is None:
+        dtype = getattr(np.zeros(1, dtype=x.dtype).sum(), "dtype", object)
+
+    if isinstance(x._meta, np.ndarray) and not isinstance(x._meta, np.matrix):
+        return x.sum(axis=axis, dtype=dtype)
+
+    def sum_drop_keepdims(*args, **kwargs):
+        kwargs.pop("computing_meta", None)
+        # masked operations on sparse produce which numpy matrices gives the same API issues handled here
```

Comment:
I don’t understand the grammar, can you elaborate?

Reply:
I apologize for the wording, it can be improved but this is the issue:

```python
import scipy.sparse as sp
import numpy as np

mat = sp.random(100,100, format="csr")
assert type(np.sum(mat, axis=0)) == np.matrix
```

Sorry for the horrifying wording, but there is little to be done, I think

Reply:
I’m aware of that. I think we should just use our sum implementation instead of the numpy one, and remove all support for `np.matrix`, which makes everything harder.

Or are there parts where `np.sum` is called outside of our control.

Reply:
What would "our implementation" look like?

```python
assert mat.sum(axis=0) == np.matrix
```

Reply:
Sorry, also the comment was correct if grammatically not so.  Doing something like `X > 2` on a sparse-meta dask arrays with yield a numpy matrix class.

Reply:
I’d like to make sure that we never have chunks that are `np.matrix` instances, since we eradicated this type from scanpy’s and anndata’s codebase long ago by using `np.asarray` in the right places (except for in some dask arrays apparently)

Reply:
The I think it's reasonable to remove it here.  Then wherever it errors once this is integrated, we can simply fix it then.